src/languages/nim/index.ts

```diff
@@ -110,15 +110,26 @@ const nimLanguage: Language = {
     addVarDeclarations,
     assertInt64,
   ],
-  detokenizer: defaultDetokenizer(
-    (a, b) =>
-      a !== "" &&
-      b !== "" &&
-      ((/[A-Za-z0-9_]/.test(a[a.length - 1]) && /[A-Za-z0-9_]/.test(b[0])) ||
-        ("=+-*/<>@$~&%|!?^.:\\".includes(a[a.length - 1]) &&
-          "=+-*/<>@$~&%|!?^.:\\".includes(b[0])) ||
-        (/[A-Za-z]/.test(a[a.length - 1]) && b[0] === `"`))
-  ),
+  detokenizer: defaultDetokenizer((a, b) => {
+    if (a === "" || b === "") return false; // glue token
```

Comment:
Explicitly using something like "$NO_WHITESPACE$" or "$GLUE_TOKEN$" rather than an empty string may make it more clear at a glance why this is being used in the emitter.